From 20d2523475523d8a913b718e39f95f90f4746483 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 25 May 2020 16:45:56 +0000
Subject: [PATCH 07/18] Disable -fsized-deallocation on macOS

The check passes on older SDKs, but the compilation still fails
SConstruct should be updated to use something similar to:
https://github.com/gperftools/gperftools/commit/98753aa7

Drop this patch when MACOSX_DEPLOYMENT_TARGET is updated to 10.12+

xref: https://jira.mongodb.org/browse/SERVER-36243
xref: https://jira.mongodb.org/browse/SERVER-40197
---
 SConstruct | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index acabb79942..05b6b65651 100644
--- a/SConstruct
+++ b/SConstruct
@@ -2962,7 +2962,10 @@ def doConfigure(myenv):
         AddToCCFLAGSIfSupported(myenv, "-Wno-unused-lambda-capture")
 
         # Enable sized deallocation support.
-        AddToCXXFLAGSIfSupported(myenv, '-fsized-deallocation')
+        # Requires macOS >=10.12
+        # Actual fix should be similar to https://github.com/gperftools/gperftools/commit/98753aa7
+        if not env.TargetOSIs('macOS'):
+            AddToCXXFLAGSIfSupported(myenv, '-fsized-deallocation')
 
         # This warning was added in Apple clang version 11 and flags many explicitly defaulted move
         # constructors and assignment operators for being implicitly deleted, which is not useful.
-- 
2.33.1

